node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java

```diff
@@ -608,8 +608,11 @@ public Node markNodeAvailableForNewAllocation(String hostname, Agent agent, Stri
                 List<Node> children = list().childrenOf(node).asList();
                 children.forEach(child -> requireRemovable(child, true, force));
                 db.removeNodes(children);
-                move(node, State.deprovisioned, Agent.system, Optional.empty());
                 List<Node> removed = new ArrayList<>(children);
+                if (zone.cloud().value().equals("aws"))
```

Comment:
Looks like we have this in several places now. We should find a way to model it properly soon.

Reply:
Agree. What I want to do is create a "cloud config" which defines the properties of a cloud and then only test for those properties.